viewer/packages/cad-styling/src/NodeIdNodeCollection.ts

```diff
@@ -0,0 +1,89 @@
+/*!
+ * Copyright 2022 Cognite AS
+ */
+import { CogniteClient, ListResponse, Node3D } from '@cognite/sdk';
+import chunk from 'lodash/chunk';
+import { CdfModelNodeCollectionDataProvider } from './CdfModelNodeCollectionDataProvider';
+import { CdfNodeCollectionBase } from './CdfNodeCollectionBase';
+import { NodeCollectionSerializationContext, registerNodeCollectionType } from './NodeCollectionDeserializer';
+import { SerializedNodeCollection } from './SerializedNodeCollection';
+
+/**
+ * Collection that holds a set of nodes including children identified by nodeIds. Note that
+ * this involves remapping IDs to "tree indices" and subtree sized used by Reveal using
+ * the CDF API. Its often better to use one of the other node collections or {@see TreeIndexNodeCollection}
+ * whenever possible for best performance.
+ */
+export class NodeIdNodeCollection extends CdfNodeCollectionBase {
+  public static readonly classToken = 'NodeIdNodeCollection';
+
+  private readonly _client: CogniteClient;
+  private readonly _model: CdfModelNodeCollectionDataProvider;
+
+  private _nodeIds: number[] = [];
+
+  constructor(client: CogniteClient, model: CdfModelNodeCollectionDataProvider) {
+    super(NodeIdNodeCollection.classToken, model);
+    this._client = client;
+    this._model = model;
+  }
+
+  /**
+   * Populates the collection with the nodes with the IDs provided. All children of
+   * the nodes are also included in the collection.
+   * @param nodeIds IDs of nodes to include in the collection.
+   * @returns Promise that resolves when the collection is populated.
+   */
+  async executeFilter(nodeIds: number[]): Promise<void> {
+    this._nodeIds = [...nodeIds];
+
+    if (nodeIds.length === 0) {
+      return this.updateCollectionFromResults([]);
+    }
+
+    const { modelId, revisionId } = this._model;
+    const executeRequest = async (batchNodeIds: number[]) => {
+      const response = await this._client.revisions3D.retrieve3DNodes(
+        modelId,
+        revisionId,
+        batchNodeIds.map(id => ({ id }))
+      );
+      const listResponse = new SingleRequestListResponse<Node3D[]>(response);
+      return listResponse;
+    };
+    const responses = chunk(nodeIds, 1000).map(chunk => executeRequest(chunk));
```

Comment:
Why smaller array, any significance with 1000 number here?

Reply:
Yes, this is a limitation in the API - see https://docs.cognite.com/api/v1/#tag/3D-Model-Revisions/operation/get3DNodesById